grails-app/conf/Config.groovy

```diff
@@ -1300,6 +1300,9 @@ openboxes.expirationDate.format = Constants.EXPIRATION_DATE_FORMAT
 openboxes.display.date.format = Constants.DISPLAY_DATE_FORMAT
 openboxes.display.date.defaultValue = Constants.DISPLAY_DATE_DEFAULT_VALUE
 
+// Notifications configuration
+openboxes.notification.disappear.time.ms = 8000
```

Comment:
We have to be careful with the word "notification" as its overloaded and can be used to mean a lot of different things (email notifications, push notifications, etc). So let's try to be more specific here. 
```
openboxes.toast...
```
Or if we need to clarify the context ...
```
openboxes.react.toast...
```

And also `disappear` and `ms` are a bit confusing to me. Disappear is actually a fairly descriptive word, I just haven't seen it used in software before. It might just be me though so feel free to state your case with some examples.

We have a convention (for good or bad) in which we include the property description and unit of measure in the property name. Something like this

- autoCloseInMilliseconds 
- autoHideInMilliseconds
- disappearInMilliseconds

I've never loved that but it was something I came up with to avoid confusion. We could also just add a descriptive comment above the property to explain what it's used for and include the UoM. Or we could choose to always standardize on milliseconds. In any of those cases, the following might be better

- openboxes.toast.autoClose
- openboxes.toast.autoHide
- openboxes.toast.autoDisappear

And instead of "time.ms" I prefer a more descriptive word like

- duration
- delay
- timeout

One other thing that we might want to configure here is whether the "toast" is actually allowed to disappear after some period of time. We probably don't need a boolean config property for the API should support enabling / disabling the auto hide so we can keep some messages on screen until the user closes them.

```
openboxes.toast.autoHide.enabled = true
openboxes.toast.autoHide.delayInMilliseconds = 5000
```
or with a comment
```
// Enable autohide of toasts by default and delay autohide in 5 seconds
openboxes.toast.autohide.enabled = true
openboxes.toast.autohide.delay = 5000
```

FWIW I'm not saying we shouldn't use the `time.ms` I just haven't personally seen that myself so I would just like to see that used commonly out in the wild in order to justify its use.

In general, I usually go to a library that we're trying to mimic and find out what their API looks like before I start naming things. 

Bootstrap uses `delay` and `autohide`
https://getbootstrap.com/docs/4.2/components/toasts/#options

React Toast uses `autoClose`
https://fkhadra.github.io/react-toastify/introduction/

jQuery Toast uses `hideAfter`
https://github.com/kamranahmedse/jquery-toast-plugin

Reply:
Another example, VueJS uses just `duration` to signify how long the toast should be displayed, implying that all toasts auto-hide unless (I assume) they have `dismissible` set to true. 
https://www.npmjs.com/package/vue-toast-notification#user-content-available-options

Not sure if that's what we want for our toast API but it would possibly remove the need for the `enabled` config property discussed above. Plus I guess the decision to make a message "always visible, but closeable" vs "visible for a certain duration" is probably a developer concern, not something that a sysadmin should be in the business of overriding. That is, unless we set the auto hide duration too low I guess.

Reply:
1. I don't feel like there is a need to keep it until tech huddle, since it is a blocker for Dariusz's ticket
2. I also didn't feel comfortable with the naming, but when naming it, I felt like I've forgotten whole English, if you know what I mean :laughing: 
3. I couldn't make it `alert()` as the library is named, because the `alert()` is already a reserved word and a built-in alert in the browser, so I discussed it with the guys and decided to go with `notification()` for now and if anyone suddenly has their own opinion going further, just point it out in the PR, so thanks for your compehensive comment about this one.
4. It's not my "invention" of having this auto hide - it was specified in the mockup that the notifications should hide after 8 seconds.
5. I will tell the guys to read your comment and we can agree on final naming for the `notification()` - as for "disappear" - I love the "duration" here, so thanks again for that research and for showing your process of dealing with such stuff.

Reply:
Great points. I'll let @awalkowiak and others (including @mdpearson) ping pong this around for a few hours tomorrow. But I agree - if this is a blocker we can merge a version we can live with for now and then make a final decision later.

Also, I hate naming. The only way I survive naming exercises is by finding someone else who has named something in a way that I either like or (at least) don't hate as much as the names I come up with on my own.

Reply:
My naming suggestion is
```
openboxes.client.notification.autohide.delay.milliseconds = 5000
```
At first I thought that we could specify the unit of measure in the comment above but I think that it would be better to explicitly have the units in the variable name to avoid any potential confusion.
That way a developer doesn't have dig through the config file looking for comments wondering if it is seconds or milliseconds.

Also I think that we should stick with name `notification` instead of `toast` since everyone has been calling it that way and I specify a prefix that it is a **client** notification

Reply:
@jmiranda @kchelstowski I like the Dariusz suggestion with `openboxes.client.notification.(...)` because I don't like using the keyword `react`.